### ISSUE TITLE
Java experiment support

### ISSUE THREAD

**Issue by budak7273** [AutomatingSciencePipeline/Monorepo#82]
- Jacob Ashworth can test with his experiment
- Users submit a compiled jar file (until zip uploading is done)
- Need to target specific java versions (this still applies to python as well, we haven't dealt with this)
- Need backend to support jar files

Time Spent
- David: 6 hrs

**Comment by budak7273** [comment]
Contact Jacob to find out details of what his experiment does right now

**Comment by SpookyBeverage** [comment]
Ok So it turns out having a docker container with multiple languages is kind of hard. There's some sort of fancy thing called multi-stage building but to be honest I have no idea how it works. Luckily docker runs in linux and you can just install java using linux, wish I would've thought of that before I spent 2 hours trying other roads

Source: https://phoenixnap.com/kb/how-to-install-java-ubuntu

**Comment by SpookyBeverage** [comment]
Yahoo! I got a jar to run on the system, was going to try and do separate config types for Java and python but it turns out they can both use the .ini files that we were already making. Had to make some minor tweaks to the run_experiment code so that there was less code duplication so it was more a refactoring.

Also I added two new dependencies, one is python-magic which looks at a files magic bytes to determine its filetype and then libmagic1 which is one of its dependencies
 
Docs for python-magic: https://pypi.org/project/python-magic/
